rust/signer/src/lib.rs

```diff
@@ -159,32 +141,35 @@ pub unsafe extern fn blockies_icon(blockies_seed: *mut StringPtr) -> *mut String
 
 // random phrase ffi
 #[no_mangle]
-pub unsafe extern fn random_phrase(words: u32) -> *mut String {
-  let words = wordlist::random_phrase(words as usize);
-  Box::into_raw(Box::new(words))
+pub unsafe extern fn random_phrase() -> *mut String {
+  use bip39::{Mnemonic, MnemonicType, Language};
+  let mnemonic = Mnemonic::new(MnemonicType::Words24, Language::English);
+  Box::into_raw(Box::new(mnemonic.into_phrase()))
 }
 
 // data encryption ffi
 
 #[no_mangle]
 pub unsafe extern fn encrypt_data(data: *mut StringPtr, password: *mut StringPtr) -> *mut String {
   let data = (*data).as_str();
-  let password = Password::from((*password).as_str());
-  let crypto = Crypto::with_plain(data.as_bytes(), &password, 10240).unwrap();
-  Box::into_raw(Box::new(crypto.into()))
+  let password = Protected::new((*password).as_str().as_bytes().to_vec());
+  let crypto = Crypto::encrypt(data.as_bytes(), &password, CRYPTO_ITERATIONS).unwrap();
+  Box::into_raw(Box::new(serde_json::to_string(&crypto).unwrap()))
```

Comment:
Might be good to have `expect`s instead of unwraps

Reply:
Unfortunately they aren't very useful here since you never see the message when the mobile app crashes.

I want to abstract out the bridge glue code so type conversions are automatic and have some way of propagating errors up to JS without panicing, then get rid of all `unwrap`s. Another thing is the duplicated logic between iOS and Android that I want to get rid of as well. That said, that will be a separate PR (likely part of #223).